app/commands/iteration/destroy.rb

```diff
@@ -6,7 +6,8 @@ class Destroy
 
     def call
       iteration.update!(deleted_at: Time.current)
-      solution.update!(published_iteration_id: nil) if solution.published_iteration_id == iteration.id
+      solution.update!(published_iteration_id: nil, published_at: nil) if solution.published_iteration_id == iteration.id
+      solution.update!(completed_at: nil) if solution.iterations.not_deleted.empty?
```

Comment:
Rather than `empty?` you might like to do `exists?` which is quicker and change to `unless`